tests/sentry/deletions/test_project.py

```diff
@@ -212,3 +223,44 @@ def test_delete_with_uptime_monitors(self):
         assert not ProjectUptimeSubscription.objects.filter(
             id=project_uptime_subscription.id
         ).exists()
+
+    def test_delete_with_workflow_engine_models(self):
+        project = self.create_project(name="test")
+
+        data_condition_group = self.create_data_condition_group()
+        data_condition = self.create_data_condition(condition_group=data_condition_group)
+        snuba_query = self.create_snuba_query()
+        subscription = QuerySubscription.objects.create(
+            project=project,
+            status=QuerySubscription.Status.ACTIVE.value,
+            subscription_id="123",
+            snuba_query=snuba_query,
+        )
+        data_source = self.create_data_source(
+            organization=self.organization, source_id=subscription.id
+        )
```

Comment:
fyi - can create these with this helper method since you're using the base workflow test: https://github.com/getsentry/sentry/blob/a17e7093f0270e4b37d38149ee06af6ad3ede7ea/tests/sentry/workflow_engine/test_base.py#L159-L197

```py
data_source, data_packet = self.create_test_query_data_source()
```
then if you want to access the snuba_query or subscription you can through the data_packet

Reply:
if I use this I can't assert that the snuba query and query subscription objects are deleted though since they're not actually created

Reply:
feel free to update the method to save it if you want -- also don't feel strongly, just an fyi for future :)